gcc/rust/typecheck/rust-hir-type-check-pattern.cc

```diff
@@ -273,5 +273,82 @@ TypeCheckPattern::visit (HIR::LiteralPattern &pattern)
 			     pattern.get_literal (), pattern.get_locus ());
 }
 
+void
+TypeCheckPattern::visit (HIR::RangePattern &pattern)
+{
+  // Resolve the upper and lower bounds, and ensure they are compatible types
+  TyTy::BaseType *upper = nullptr, *lower = nullptr;
+
+  // It would be nice to factor this out into a helper since the logic for both
+  // bounds is exactly the same...
```

Comment:
```suggestion
  // TODO: It would be nice to factor this out into a helper since the logic for both
  // bounds is exactly the same...
```
Or FIXME, whichever you like more. But I think that makes it easier to find and fix tiny issues like those